src/main/java/org/codehaus/plexus/archiver/zip/AbstractZipArchiver.java

```diff
@@ -534,35 +536,28 @@ public InputStream get()
         }
     }
 
-    protected void setTime( ZipArchiveEntry zipEntry, long lastModified )
+    /**
+     * Set the ZipEntry dostime using the date if specified otherwise the original time.
+     *
+     * <p>Zip archives store file modification times with a granularity of two seconds, so the times will either be
+     * rounded up or down. If you round down, the archive will always seem out-of-date when you rerun the task, so the
+     * default is to round up. Rounding up may lead to a different type of problems like JSPs inside a web archive that
+     * seem to be slightly more recent than precompiled pages, rendering precompilation useless.
+     * plexus-archiver chooses to round up.
+     *
+     * @param zipEntry to set the last modified time
+     * @param lastModifiedTime to set in the zip entry only if {@link #getLastModifiedTime()} returns null
+     */
+    protected void setZipEntryTime( ZipArchiveEntry zipEntry, long lastModifiedTime )
     {
         if ( getLastModifiedTime() != null )
         {
-            lastModified = getLastModifiedTime().toMillis();
+            lastModifiedTime = getLastModifiedTime().toMillis();
         }
 
-        // Zip archives store file modification times with a
-        // granularity of two seconds, so the times will either be rounded
-        // up or down. If you round down, the archive will always seem
-        // out-of-date when you rerun the task, so the default is to round
-        // up. Rounding up may lead to a different type of problems like
-        // JSPs inside a web archive that seem to be slightly more recent
-        // than precompiled pages, rendering precompilation useless.
-        // plexus-archiver chooses to round up.
-        zipEntry.setTime( lastModified + 1999 );
-
-        /*   Consider adding extended file stamp support.....
-
-         X5455_ExtendedTimestamp ts =  new X5455_ExtendedTimestamp();
-         ts.setModifyJavaTime(new Date(lastModified));
-         if (zipEntry.getExtra() != null){
-         // Uh-oh. What do we do now.
-         throw new IllegalStateException("DIdnt expect to see xtradata here ?");
-
-         }  else {
-         zipEntry.setExtra(ts.getLocalFileDataData());
-         }
-         */
+        // Round up 2 seconds when needed
+        lastModifiedTime = ( lastModifiedTime + 1999 ) - ( ( lastModifiedTime + 1999 ) % 2000 );
```

Comment:
Looks like this is change in how `setTime` works, not only method rename as the commit suggest, or I'm not reading the diff correctly?

Reply:
I don't remember right now why I made this change, but it actually preserve the current behavior and it "normalizes" the time by truncating the millis.

Anyway, if you think that this is weird I could just simply remove it and leave it as before.

Reply:
Reverted this line and also improved a bit the test `testLastModifiedTimeRounding`.